### ISSUE TITLE
Not able to publish model with custom name

### ISSUE THREAD

**Issue by pkaczynski** [sassoftware/python-sasctl#131]
publish_model contains name argument but it is not used later on. You can not publish a model with a custom publishing name. There's `model_name` calculated but not used in request.

https://github.com/sassoftware/python-sasctl/blob/a696db8d74882af314308503313c32abffd60e68/src/sasctl/_services/model_management.py#L29

**Comment by smlindauer** [comment]
It looks like there are duplicate publish_model functions across services. I'd suggest using the function found in the model_publish service: https://github.com/sassoftware/python-sasctl/blob/a696db8d74882af314308503313c32abffd60e68/src/sasctl/_services/model_publish.py#L94

I will take a look at why we have this duplication and reassess the model_management service's implementation of publish_model.

**Comment by pkaczynski** [comment]
Model publish service function you refer to does not have an option to force and overwrite published model... 

On the other hand I get an error:

```
C:\devel\venvs\pzu\lib\site-packages\urllib3\connectionpool.py:1020: InsecureRequestWarning: Unverified HTTPS request is being made to host 'rfnk01-0078.exnet.sas.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
Traceback (most recent call last):
  File "***********/AppData/Roaming/JetBrains/PyCharm2021.3/scratches/publish.py", line 9, in <module>
    res = mp.publish_model("GRADBOOST", destination='maslocal')
  File "***************\lib\site-packages\sasctl\_services\model_publish.py", line 127, in publish_model
    name=name or model.get("name"),
AttributeError: 'NoneType' object has no attribute 'get'
```
Seems like this line:
```
model_uri = cls._model_repository.get_model_link(model, "cls")
```
Does not return anything that has "cls" in links.

**Comment by smlindauer** [comment]
Hey @pkaczynski, 

Apologies on the delay in response, getting the most recent release of sasctl uncovered a number of SAS Viya 4 API bugs that needed to be stamped out one at a time.

I believe the error you are seeing was due to an incorrectly applied filter (due to a previous error with modelRepository on SAS Viya 3.4) on the get_model() call in the line above. It should have been fixed in 9df0eee76456d3d35c4d7b414a833293e6da8ac9 and is out on the most recent release.

Returning to the core of your question, I am working on a fix for the model_management publish_model call to properly allow for custom publish names. As you stated, it looks like the name argument is not properly handled within the function. In the meantime, you should be able to use the model_publish publish_model call again (but obviously that will lack the overwrite capacity that the model_management call has).
